SalesforceLiveAgentApp.ts

```diff
@@ -0,0 +1,261 @@
+import {
+  IAppAccessors,
+  IConfigurationExtend,
+  IEnvironmentRead,
+  IHttp,
+  ILogger,
+  IModify,
+  IPersistence,
+  IRead,
+} from '@rocket.chat/apps-engine/definition/accessors';
+import { App } from '@rocket.chat/apps-engine/definition/App';
+import {
+  ILivechatMessage,
+  ILivechatRoom,
+} from '@rocket.chat/apps-engine/definition/livechat';
+import {
+  IMessage,
+  IPostMessageSent,
+} from '@rocket.chat/apps-engine/definition/messages';
+import {
+  IAppInfo,
+  RocketChatAssociationModel,
+  RocketChatAssociationRecord,
+} from '@rocket.chat/apps-engine/definition/metadata';
+import { IUser } from '@rocket.chat/apps-engine/definition/users';
+import { AppSettings } from './AppSettings';
+import { InitiateSalesforceSession } from './handlers/InitiateSalesforceSession';
+import {
+  sendDebugLCMessage,
+  sendLCMessage,
+} from './helperFunctions/GeneralHelpers';
+import { SalesforceHelpers } from './helperFunctions/SalesforceHelpers';
+
+export class SalesforcePluginApp extends App implements IPostMessageSent {
+  constructor(info: IAppInfo, logger: ILogger, accessors: IAppAccessors) {
+    super(info, logger, accessors);
+  }
+
+  public async initialize(
+    configurationExtend: IConfigurationExtend,
+    environmentRead: IEnvironmentRead,
+  ): Promise<void> {
+    await this.extendConfiguration(configurationExtend);
+    this.getLogger().log('App Initialized');
+  }
+
+  public async executePostMessageSent(
+    message: IMessage,
+    read: IRead,
+    http: IHttp,
+    persistence: IPersistence,
+    modify: IModify,
+  ): Promise<void> {
+    const dialogflowBotUsername: string = (
+      await read
+        .getEnvironmentReader()
+        .getSettings()
+        .getById('dialogflow_bot_username')
+    ).value;
+    const salesforceBotUsername: string = (
+      await read
+        .getEnvironmentReader()
+        .getSettings()
+        .getById('salesforce_bot_username')
+    ).value;
+    const salesforceChatApiEndpoint: string = (
+      await read
+        .getEnvironmentReader()
+        .getSettings()
+        .getById('salesforce_chat_api_endpoint')
+    ).value;
+
+    if (message.sender.username === dialogflowBotUsername) {
+      return;
+    } else if (message.room.type !== 'l') {
+      return;
+    }
+
+    const lmessage: ILivechatMessage = message;
+    const lroom: ILivechatRoom = lmessage.room as ILivechatRoom;
+    const LcAgent: IUser = lroom.servedBy ? lroom.servedBy : message.sender;
+
+    if (message.text === 'initiate_salesforce_session') {
+      const initiateSalesforceSessionhandler = new InitiateSalesforceSession(
+        message,
+        read,
+        http,
+        persistence,
+        modify,
+      );
+
+      try {
+        initiateSalesforceSessionhandler.exec();
+      } catch (error) {
+        console.log(error);
+      }
+    }
+
+    if (LcAgent.username === salesforceBotUsername) {
```

Comment:
In order to review your code, I’ll need to wait until you remove all of these break lines.